http-client/Network/HTTP/Client/Request.hs

```diff
@@ -111,16 +113,30 @@ getUri req = URI
     , uriFragment = ""
     }
 
+applyAnyUriBasedAuth :: URI -> Request -> Request
+applyAnyUriBasedAuth uri =  applyBasicAuth (username authInfo) (password authInfo)
+ where
+   authInfo = uriUserInfo $ fromJust $ uriAuthority uri
```

Comment:
I'm not familiar with this part of `network-uri`. Under what circumstances is `uriAuthority` a `Just` value, and when would it be `Nothing`?